src/commands/giveaway.rs

```diff
@@ -0,0 +1,650 @@
+            }
+            "GIVEAWAY_list_next" => {
+                let offset = offsets.get(&component.user.id.0).unwrap_or(&0).to_owned() + 10;
+                let embeds = generate_list_embeds(&db, offset).await;
+                let components = generate_list_components(offset, embeds.len() as i64);
+                debug!(
+                    "Showing next {} giveaways for user {}",
+                    embeds.len(),
+                    component.user.id.0
+                );
+                component
+                    .create_interaction_response(&ctx.http, |r| {
+                        r.kind(InteractionResponseType::UpdateMessage);
+                        r.interaction_response_data(|d| {
+                            d.set_components(components);
+                            d.embeds(embeds)
+                        })
+                    })
+                    .await
+                    .unwrap();
+                offsets.insert(component.user.id.0, offset);
+            }
+            _ => debug!("Unknown interaction: {}", component.data.custom_id),
+        }
+    }
+}
+
+pub async fn handle_interaction(ctx: &Context, interaction: ApplicationCommandInteraction) {
+    let db = ctx.get_db().await;
+
+    ensure_offset_map(&ctx).await;
+    let mut data = ctx.data.write().await;
+    let offset_map = data.get_mut::<ListOffset>().unwrap();
+
+    let option = interaction
+        .data
+        .options
+        .first()
+        .expect("Giveaway subcommand missing");
+    let sub_options = option.options.clone();
+
+    match option.name.as_str() {
+        "start" => {
+            let channel = sub_options
+                .by_name("channel")
+                .expect("Missing channel option")
+                .to_channel()
+                .expect("Invalid channel option");
+            let duration = sub_options
+                .by_name("duration")
+                .map_or(DEFAULT_DURATION, |x| x.to_i64().unwrap_or(DEFAULT_DURATION));
+            let winners = sub_options
+                .by_name("winners")
+                .map_or(DEFAULT_WINNERS, |x| x.to_i64().unwrap_or(DEFAULT_WINNERS));
+            let prize = sub_options
+                .by_name("prize")
+                .map_or(DEFAULT_PRIZE.to_string(), |x| {
+                    x.to_string().unwrap_or(DEFAULT_PRIZE.to_string())
+                });
+
+            if winners < 1 || duration < 1 {
+                interaction
+                    .create_interaction_response(&ctx.http, |r| {
+                        r.interaction_response_data(|d| {
+                            d.content(
+                                "Duration and winners must be positive integers or left empty.",
+                            );
+                            d.flags(InteractionApplicationCommandCallbackDataFlags::EPHEMERAL)
+                        })
+                    })
+                    .await
+                    .unwrap();
```

Comment:
Maybe just have duration=0 be infinite here.

Reply:
This should be implemented in a future PR, this one already has a lot of features in it.